### PR TITLE
Feat: add error report mechanism

### PR DESCRIPTION
**Pull request by igorsmotto**
## Proposed Changes
- Add an error report mechanism: Every time an exception is detected by `picocli` it will send the exception details + the command executed to `api.mobile.dev`. 
    - This mechanism tries to not collect any PII information.
    - `-e/--env` can have PII information, so this PR also hash the values with SHA-256.
    - Also, there is a timeout of 1 second to complete the request to make sure that the user experience is not affected.
- Add `SystemInformationInterceptor` to `ApiClient` to simplify adding system information for each request.

### Testing
Tested manually:
- Modified an existing maestro command to always throw a CLIError
- Verified if the request went through

### Further possible improvements
- Send a _category_ of error. This could be made easier if there are custom exceptions for each _category_.
- Prompt the user to send all logs (saved by `DebugLogStore`) and flow file/workspace.

### CODE REVIEW

File: maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import maestro.cli.api.ApiClient
+import picocli.CommandLine
+import java.security.MessageDigest
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import kotlin.Exception
+
+object ErrorReporter {
+
+    private const val BASE_API_URL = "https://api.mobile.dev"
```
Comment (axelniklasson) [inline maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt:12]: Nit: we're defining `BASE_API_URL` here and also in `Updates.kt`. Shall we define it once somewhere and use it everywhere instead?

File: maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                key + "=" + hashString(value)
+            } else arg
+        }
+
+        executor.submit {
+            ApiClient(BASE_API_URL).sendErrorReport(
+                exception,
+                scrubbedArgs.joinToString(" ")
+            )
+        }.get(1, TimeUnit.SECONDS)
```
Comment (axelniklasson) [inline maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt:31]: What happens if this request fails? Will there be an error printed to users or will it be swallowed and hidden from them?

**Review by axelniklasson — APPROVED** [review]

File: maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt (hunk 2 above)
Comment (igorsmotto) [reply maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt]: I believe all of this reporting stuff should be invisible to the user, as this is not the primary concern of the user when running maestro. Fix here: https://github.com/mobile-dev-inc/maestro/pull/584/commits/979370c1cd94d25720f5fc88ee2b4b11db1f18d9

Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/util/ErrorReporter.kt]: Excellent, I completely agree 🙌

**Review by axelniklasson — APPROVED** [review]

**Pull request merged**